CRM/Contribute/BAO/Contribution.php

```diff
@@ -4145,6 +4146,27 @@ public static function getPaymentInfo($id, $component, $getTrxnInfo = FALSE, $us
           }
           $paidByLabel .= " ({$creditCardType}{$pantruncation})";
         }
+
+        // show payment edit link only for payments done via backoffice form
```

Comment:
Smart debit for direct debits in the UK certainly supports changing the value of the transaction via the API, it's not really being used yet on the civi side, but you can do it via the "Update billing details" link.